src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -523,6 +526,13 @@ private ZipEntry readLOC() throws IOException {
         }
         e.method = get16(tmpbuf, LOCHOW);
         e.xdostime = get32(tmpbuf, LOCTIM);
+
+        // Expect 32-bit Data Descriptor size fields by default
+        expect64BitDataDescriptor = false;
+
+        long csize = get32(tmpbuf, LOCSIZ);
+        long size = get32(tmpbuf, LOCLEN);
```

Comment:
Hello Eirik, I suspect this part of the change has an issue. Before reading the `tmpbuf` for compressed and uncompressed sizes, there will be 32 bits of CRC, which should be read first. This now skips those 32 CRC bits and reads them (in the else block) after reading these sizes and that can cause incorrect LOC data.

Reply:
The Github actions job which runs tier1 is all successful with this proposed change. So I'm a bit surprised that the tests didn't catch any issues, which makes me wonder if we have enough test coverage that covers this change.

Reply:
> Hello Eirik, I suspect this part of the change has an issue. Before reading the `tmpbuf` for compressed and uncompressed sizes, there will be 32 bits of CRC, which should be read first. This now skips those 32 CRC bits and reads them (in the else block) after reading these sizes and that can cause incorrect LOC data.

How does the order of the reads from the byte array matter? Outside cache efficiency I would presume they could be read in any order?

(These reads are from a temp buffer, not the stream)

Could you elaborate? I'm not sure I'm following :-)

Reply:
>(These reads are from a temp buffer, not the stream)

Ah! you are right indeed. I didn't correctly read that part of that code. It's reading from a temp buffer which has been fully initialized with a LOC and these reads happens with specific offsets within that buffer. So yes, you are correct that the order won't matter here. Thank you for that detail.